### PR TITLE
chore: fix codecov

### PR DESCRIPTION
**Pull request by violenil**
Partly completes https://github.com/jina-ai/now.private/issues/313

Problem definition:

Uploading the coverage files to codecov was removed from CD in [this commit](https://github.com/jina-ai/now/commit/11a334196e5ca35c1777a3e1e324b7e633f65bd7), which is why the commit we have been comparing all of our PRs on is [this one](https://github.com/jina-ai/now/commit/ea7eab1633c95a8fbfbb4b14a87eb36a742b8b63), committed just before and last to run cd with codecov upload.

This PR simplifies the ci codecov logic and also adds the codecov upload back into the cd.

---

- [x] This PR references an open issue
- [ ] ~Added a test~
- [ ] ~Updated the documentation~
- [ ] ~Added a screenshot of a successful customer deployment~

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -222,22 +232,6 @@ jobs:
...
-          python -m pip install --upgrade pip
-          pip install pytest-cov
-      - name: Download all coverage artifacts
-        uses: actions/download-artifact@v2
-      - name: Run coverage
-        run: |
-          coverage combine coverage*/.coverage*
-          coverage xml
-      - name: Upload coverage from test to Codecov
-        uses: codecov/codecov-action@v2
```
Comment (joschkabraun) [inline .github/workflows/ci.yml:240]: shouldn't we need to upload the combined ones in CI if we upload the single ones?

**Review by joschkabraun — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (violenil) [reply .github/workflows/ci.yml:240]: No, codecov takes care of this. We do not need to download+combine+push.

Comment (joschkabraun) [reply .github/workflows/ci.yml:240]: ok, great!

**Review by joschkabraun — APPROVED** [review]
LGTM!

**Review by azayz — APPROVED** [review]

**Review by Tanguyabel — APPROVED** [review]
lgtm

**Pull request merged**